Open gles upgrade

ThorstenBux opened the pull request.

BoiseBaked commented on AndroidStudioProjects/ARBaseLibProj/.idea/compiler.xml:
```diff
@@ -11,6 +11,7 @@
       <entry name="!?*.flex" />
```
I don't think we want any of the files under an .idea directory versioned. I can get you an effective .gitignore entry that will globally ignore all content under the .idea directory recursively. These file might have been inadvertently versioned originally. The have to be unversioned (git rm) and then ignored (.gitignore).

ThorstenBux replied:
done in the master branch, will merge to this branch

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.gitignore:
```diff
@@ -0,0 +1,8 @@
+*.iml
```
Need to update the .gitignore file under AndroidStudioProjects directory with the following .gitignore entry: /**/.idea

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/compiler.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/copyright/profiles_settings.xml:
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/encodings.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/gradle.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/modules.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/.idea/runConfigurations.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Add the /**/.idea to AndroidStudioProjects/.gitignore file.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/gradlew.bat:
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
Looks like you like to run Android Studio on Windows. In that case please make sure you configure "core.autocrlf=true" on your Windows devbox.
To see if or what you have core.autocrlf set to, do: "git config --global --list" and look for "core.autocrlf=true." On OS X devboxes, please set: "core.autocrlf=input."

ThorstenBux replied:
done

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/build.gradle:
```diff
@@ -0,0 +1,53 @@
+
+model {
+    android {
+        compileSdkVersion = 22
```
I think I saw in another build.gradle file the compileSdkVersion was set to 23. 22 or 23 is fine but please be consistent.

ThorstenBux replied:
ok

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/marker.dat:
```diff
@@ -0,0 +1,44 @@
+#the number of patterns to be recognized
```
Maybe in should indicate to git to treat all .dat files as binary files since these are pretty much un-merge-able.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/patt.a:
```diff
@@ -0,0 +1,196 @@
+ 201 211 215 214 217 221 222 219 198 223 223 225 226 226 225 225
```
Maybe we should indicate to git to treat all .a files as binary files since these are pretty much un-merge-able.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/patt.c:
```diff
@@ -0,0 +1,196 @@
+ 141 153 152 152 150 151 154 154 150 153 140 144 146 136 147 120
```
We probably need to establish a common file extension for pattern data files so they can essentially be indicated as binary (un-merge-able) binary files.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/patt.d:
```diff
@@ -0,0 +1,196 @@
+  96 131 130 134 123 130 136 130 133 125 134 118 125 125 129 119
```
Ditto as previous.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/patt.f:
```diff
@@ -0,0 +1,196 @@
+ 123 136 143 142 151 150 142 144 148 149 149 148 150 147 149 129
```
Ditto as previous.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/multi/patt.g:
```diff
@@ -0,0 +1,196 @@
+ 145 150 162 167 159 167 161 157 159 169 165 155 156 157 152 145
```
Ditto as previous.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/patt.hiro:
```diff
@@ -0,0 +1,196 @@
+ 234 235 240 233 240 234 240 235 240 237 240 238 240 240 240 232
```
Ditto as previous.

BoiseBaked commented on AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/patt.kanji:
```diff
@@ -0,0 +1,196 @@
+ 214 225 240 225 214 240 216 204 214 227 181 192 198 192 181 192
```
Ditto as previous.

ThorstenBux replied:
added all these files to the .gitattributes file in the AndroidStudioProjects directory.
Shall we consider this for all projects eventually? Because these files are un-merge-able independent of the project.

BoiseBaked replied:
Yes, we should establish a .gitattributes file for the root of each repo.
Especially for patt. or .patt files.

JWolf

On Sun, Jan 24, 2016 at 3:36 PM, ThorstenBux <notifications@github.com>
wrote:

> In
> AndroidStudioProjects/ARSimpleOpenGLES20/simpleOpenGlES20/src/main/assets/Data/patt.kanji
> <https://github.com/artoolkit/artoolkit5/pull/41#discussion_r50643900>:
>
> > @@ -0,0 +1,196 @@
> > + 214 225 240 225 214 240 216 204 214 227 181 192 198 192 181 192
>
> added all these files to the .gitattributes file in the
> AndroidStudioProjects directory.
> Shall we consider this for all projects eventually? Because these files
> are un-merge-able independent of the project.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41/files#r50643900>.
>

ThorstenBux replied:
Noted and will do while changing the pattern names to *.patt

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/.name:
```diff
@@ -0,0 +1 @@
+ArMarkerDistance
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/compiler.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/copyright/profiles_settings.xml:
```diff
@@ -0,0 +1,3 @@
+<component name="CopyrightManager">
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/encodings.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/gradle.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/modules.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/.idea/runConfigurations.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
See previous comments for .gitignore'ing all files under .idea recursively.

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/SampleArMarkerDistance/.gitignore:
```diff
@@ -0,0 +1 @@
+/build
```
This can be done for all projects under AndroidStudioProjects but adding "/**/build" to .../AndroidStudioProjects/.gitignore

ThorstenBux replied:
done

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/SampleArMarkerDistance/src/main/assets/Data/awesome.patt:
```diff
@@ -0,0 +1,196 @@
+ 124 127 126 124 124 121 121 121 121 120 120 120 120 119 117 117
```
Is the ".patt" filename extension supported for pattern data file? Wally once told me no. If he's right, we should make it so and support the ".patt" filename extension. That we can designate in git that such files should be treated as binary files (because they're un-merge-able).

ThorstenBux replied:
Added *.patt to the .gitattributes file. 

For me the *.patt extension works fine that is why I would like to stick to it. Do you have any final information from Wally?
Once we know for sure that .patt is working I can start change the pattern names in all projects if needed.

BoiseBaked replied:
Yes, if ".patt" works, by all means keep it, and convert any and all
example apps that use patt.[name] prefix to use a ".patt" filename
extension instead. Then we can fix the documentation to be consistent and
to use the ".patt" filename extension as the standard.

JWolf

On Sun, Jan 24, 2016 at 3:40 PM, ThorstenBux <notifications@github.com>
wrote:

> In
> AndroidStudioProjects/ArMarkerDistance/SampleArMarkerDistance/src/main/assets/Data/awesome.patt
> <https://github.com/artoolkit/artoolkit5/pull/41#discussion_r50643949>:
>
> > @@ -0,0 +1,196 @@
> > + 124 127 126 124 124 121 121 121 121 120 120 120 120 119 117 117
>
> Added *.patt to the .gitattributes file.
>
> For me the *.patt extension works fine that is why I would like to stick
> to it. Do you have any final information from Wally?
> Once we know for sure that .patt is working I can start change the pattern
> names in all projects if needed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41/files#r50643949>.
>

ThorstenBux replied:
I'll open a new branch for this then.
Or how would you prefer that I realize it?

Shall I do this for all example? AndroidStudio/Eclipse/VisualStudio (Can't do iOS and Mac right now, still waiting for my MacBook)

BoiseBaked replied:
I'll open a new branch for this then.
Or how would you prefer that I realize it?


Not sure what you mean by "realize" but hey whatever is simpler for you.

JWolf

On Tue, Jan 26, 2016 at 2:07 PM, ThorstenBux <notifications@github.com>
wrote:

> In
> AndroidStudioProjects/ArMarkerDistance/SampleArMarkerDistance/src/main/assets/Data/awesome.patt
> <https://github.com/artoolkit/artoolkit5/pull/41#discussion_r50899949>:
>
> > @@ -0,0 +1,196 @@
> > + 124 127 126 124 124 121 121 121 121 120 120 120 120 119 117 117
>
> I'll open a new branch for this then.
> Or how would you prefer that I realize it?
>
> Shall I do this for all example? AndroidStudio/Eclipse/VisualStudio (Can't
> do iOS and Mac right now, still waiting for my MacBook)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41/files#r50899949>.
>

ThorstenBux replied:
Just trying to understand how you work with branches and merging. Maybe we can have a short discussion on that after our planning? Might be easier for me to get started :)

BoiseBaked commented on AndroidStudioProjects/ArMarkerDistance/SampleArMarkerDistance/src/main/assets/Data/patt.hiro:
```diff
@@ -0,0 +1,196 @@
+ 234 235 240 233 240 234 240 235 240 237 240 238 240 240 240 232
```
Change this file and all code that accesses it to hiro.patt. Might be a big project (across the repo).

BoiseBaked commented:
We don't want to version library files (.a, .so, .dll, etc). Please remove from versioning. Also, we have really establish this standard but in the root directory of every Android Studio project, a readme.md should be added instructing developers on how to build and deploy the projects.

ThorstenBux commented:
You mean I should remove the .so files from the jniLibs directory? 
I just copied my project from the ARSimple example and there everything is versioned too.
Removing them would mean to write a detailed description how to build the project and we would need to find a place were we provide the .so files.

Also it means to remove the .aar files from each project and tell the developers to build them using the ARBase project prior to toy around with any example. 
Are you sure that we want this?

ThorstenBux commented:
I propose that once these changes are merged I can take care of the .idea files inside the other projects.
I was anyways wondering why to maintain Eclipse and AndroidStuido projects separately instead of just committing the sources and writing some instructions on how to get the projects either in Eclipse or in AndroidStudio

ThorstenBux commented:
I think I made all the changes you proposed. It is getting to be a lot of changes now so I hope we can proceed merging and do other stuff from the new master base then.

BoiseBaked commented:
On Sun, Jan 24, 2016 at 3:45 PM, ThorstenBux <notifications@github.com>
wrote:

> You mean I should remove the .so files from the jniLibs directory?
>
Yes. Everytime a user wants to build the app you want to build it's
dependencies and then build the app.

But here is the scoop. I have not converted the build.bat, the build script
for Windows desktop, similar to what I converted Mac OS X build.sh to do.
If you view what [repo]/android/build.sh, it deploys ARTK NDK dependencies
to the "libs" directory under both Eclipse and Android Studio projects. Why
the "libs" directory? Because the built libraries are place in a "libs"
directory by the Android NDK ".mk" makefiles. I also added script code to
the Android Studio build.gradle file that tells Android Studio to use the
"libs" directory instead of the "jniLibs" directory. The build.bat script
file needs to work like the build.sh script file. If you're already setup
on Windows with Android Studio, go for it and make the build.bat script
changes. My Windows 10 devbox needs to get Visual Studio 2015 before I can
do anything.

The fact that ARSimple, and probably other AS projects, has checked in
binary library files is probably because they haven't been cleaned up yet.

> I just copied my project from the ARSimple example and there everything is
> versioned too.
> Removing them would mean to write a detailed description how to build the
> project and we would need to find a place were we provide the .so files.
>
> Also it means to remove the .aar files from each project and tell the
> developers to build them using the ARBase project prior to toy around with
> any example.
> Are you sure that we want this?
>
Until Android Studio matures a little and while we still somewhat support
Eclipse, we  want to continue to build NDK-based dependencies manually. We
also want to maintain one source code base for aRBaseLib so if you have a
better way for AS projects to depend and include an external library
without requiring a directory source under multiple AS projects, let me
know.

Thanks for the good questions
JWolf

BoiseBaked commented:
Well I don't know. Eclipse is so strange and different and Android Studio
was trying to mimic Visual Studios and
IntelliJ IDEA and I wanted to setup something similar to Eclipse project.
Also I was hoping that sometime soon down the road, we could delete the
Eclipse projects directories from the ARTK repos.



On Sun, Jan 24, 2016 at 5:25 PM, ThorstenBux <notifications@github.com>
wrote:

> I propose that once these changes are merged I can take care of the .idea
> files inside the other projects.
> I was anyways wondering why to maintain Eclipse and AndroidStuido projects
> separately instead of just committing the sources and writing some
> instructions on how to get the projects either in Eclipse or in
> AndroidStudio
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41#issuecomment-174356267>.
>

BoiseBaked commented:
By all means, we can do one off fix ups to the public ARTK5 master
repo/branch before they're are pushed to the private ARTK5_dev repo. This
would be much preferred.

JWolf

On Sun, Jan 24, 2016 at 5:28 PM, ThorstenBux <notifications@github.com>
wrote:

> I think I made all the changes you proposed. It is getting to be a lot of
> changes now so I hope we can proceed merging and do other stuff from the
> new master base then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41#issuecomment-174356427>.
>

ThorstenBux commented:
>By all means, we can do one off fix ups to the public ARTK5 master repo/branch before they're are pushed to the private ARTK5_dev repo. This would be much preferred.

Sorry I do not understand what you are trying to tell me.

ThorstenBux commented:
> Well I don't know. Eclipse is so strange and different and Android Studio was trying to mimic Visual Studios and IntelliJ IDEA and I wanted to setup something similar to Eclipse project. Also I was hoping that sometime soon down the road, we could delete the Eclipse projects directories from the ARTK repos.

Well I'm used to Eclipse for Java developing but for Android the AndroidStudio seams to be way better then Eclipse with all the plugins. I worked with AndroidStudio in the first versions too. There I would have said differently but as it is now I prefer working with AndroidStudio.

BoiseBaked commented:
I believed this was referring to adding a global repo .gitattribute file to
indicate to git what should be considered as binary files for all ARTK
repos. You can do it within the big pull request or just create a separate
branch. I think a separate branch would be better. And, if you're too busy,
I can do it.

On Tue, Jan 26, 2016 at 2:44 PM, ThorstenBux <notifications@github.com>
wrote:

> By all means, we can do one off fix ups to the public ARTK5 master
> repo/branch before they're are pushed to the private ARTK5_dev repo. This
> would be much preferred.
>
> Sorry I do not understand what you are trying to tell me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/artoolkit/artoolkit5/pull/41#issuecomment-175246324>.
>

ThorstenBux commented:
Not to busy right now, ;) and might be good to get to know all the samples. I'll create a new branch from my fork and do the clean-up there.

Just a bit concerned as I have several branches right now:
* master: Containing the distanceSample and changes to ARBaseLib
* OpenGLES: branched from my master; containing additions to the ARBaseLib for OpenGL ES 2.0 support and an OpenGLESSimpleSample
* OpenGLESDistanceSample: <InWork> branched from OpenGLES; branch were I currently try to get the DistanceSample to work with ES 2.0
* CleanUp branch: todo for all the clean-up things

>I believed this was referring to adding a global repo .gitattribute file to
indicate to git what should be considered as binary files for all ARTK
repos. You can do it within the big pull request or just create a separate
branch. I think a separate branch would be better. And, if you're too busy,
I can do it.
…

>>On Tue, Jan 26, 2016 at 2:44 PM, ThorstenBux <notifications@github.com> wrote:
 By all means, we can do one off fix ups to the public ARTK5 master
 repo/branch before they're are pushed to the private ARTK5_dev repo. This
 would be much preferred.

 >>Sorry I do not understand what you are trying to tell me.

BoiseBaked commented:
On Tue, Jan 26, 2016 at 2:47 PM, ThorstenBux <notifications@github.com>
wrote:

> Well I don't know. Eclipse is so strange and different and Android Studio
> was trying to mimic Visual Studios and IntelliJ IDEA and I wanted to setup
> something similar to Eclipse project. Also I was hoping that sometime soon
> down the road, we could delete the Eclipse projects directories from the
> ARTK repos.
>
> Well I'm used to Eclipse for Java developing but for Android the
> AndroidStudio seams to be way better then Eclipse with all the plugins. I
> worked with AndroidStudio in the first versions too. There I would have
> said differently but as it is now I prefer working with AndroidStudio.
>
dev.android.com cutoff support for Eclipse for Android development in June
of last year. So it's either IntelliJ IDEA or Android Studio.
dev.android.com recommends Android Studio so we will go with Android Studio
even though NDK support isn't quite there yet.

JWolf

ThorstenBux commented:
>Yes. Everytime a user wants to build the app you want to build it's
dependencies and then build the app.

>But here is the scoop. I have not converted the build.bat, the build script
for Windows desktop, similar to what I converted Mac OS X build.sh to do.
If you view what [repo]/android/build.sh, it deploys ARTK NDK dependencies
to the "libs" directory under both Eclipse and Android Studio projects. Why
the "libs" directory? Because the built libraries are place in a "libs"
directory by the Android NDK ".mk" makefiles. I also added script code to
the Android Studio build.gradle file that tells Android Studio to use the
"libs" directory instead of the "jniLibs" directory. The build.bat script
file needs to work like the build.sh script file. If you're already setup
on Windows with Android Studio, go for it and make the build.bat script
changes. My Windows 10 devbox needs to get Visual Studio 2015 before I can
do anything.

There is my first gap in understanding. I didn't know there is something like a build script. Also I think I saw jniLibs directory used somewhere that I why I continued using it. But I can switch to libs instead. 

Where can I find the Android NDK .mk files? I looked around but could not find the project.

I have set-up VS 2015 but I'm not able to compile things because one project cannot be converted to VS2015. There I need some help and a license too ;) as I'm using my private one currently.

Bottom line for now if fear is that I cannot set-up the .bat file till I have VS2015 running as well. 

>The fact that ARSimple, and probably other AS projects, has checked in
binary library files is probably because they haven't been cleaned up yet.
>> I just copied my project from the ARSimple example and there everything is
 versioned too.
 >>Removing them would mean to write a detailed description how to build the
 project and we would need to find a place were we provide the .so files.

>> Also it means to remove the .aar files from each project and tell the
 developers to build them using the ARBase project prior to toy around with
 any example.  Are you sure that we want this?

>Until Android Studio matures a little and while we still somewhat support
Eclipse, we  want to continue to build NDK-based dependencies manually. We
also want to maintain one source code base for aRBaseLib so if you have a
better way for AS projects to depend and include an external library
without requiring a directory source under multiple AS projects, let me
know.

The way it is done currently is good. I was just trying to understand how things are done. 
Is there some kind of documentations in regards to the build and make scripts? And how to set-up AS projects besides how I did it (Just checkout import and run).

Do you think it is possible to enhance the build script to build aRBaseLib as well and copy the lib into each sample project? Or is this done already?

ThorstenBux commented:
Perfect :+1: 
>dev.android.com cutoff support for Eclipse for Android development in June of last year. So it's either IntelliJ IDEA or Android Studio. dev.android.com recommends Android Studio so we will go with Android Studio even though NDK support isn't quite there yet.
